x/merkledb/proof_test.go

```diff
@@ -22,105 +22,194 @@ import (
 )
 
 func Test_Proof_Empty(t *testing.T) {
-	proof := &Proof{}
-	err := proof.Verify(context.Background(), ids.Empty, 4, DefaultHasher)
-	require.ErrorIs(t, err, ErrEmptyProof)
+	tests := []struct {
+		name   string
+		proof  *Proof
+		expect error
+	}{
+		{
+			name:   "empty proof",
+			proof:  &Proof{},
+			expect: ErrEmptyProof,
+		},
+		{
+			name:   "empty path",
+			proof:  &Proof{Path: []ProofNode{}},
+			expect: ErrEmptyProof,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+
+			err := tt.proof.Verify(context.Background(), ids.Empty, 4, DefaultHasher)
+			require.ErrorIs(err, tt.expect)
+		})
+	}
 }
 
-func Test_Proof_Simple(t *testing.T) {
-	require := require.New(t)
+func Test_Proof_Exclusion(t *testing.T) {
+	tests := []struct {
+		name   string
+		modify func(*Proof)
```

Comment:
I would just write these as two separate tests - I think these `modify` functions (or any setup functions that pass around arguments in test tables for that matter) end up turning the test table into a test anyways.